redbot/cogs/downloader/downloader.py

```diff
@@ -1056,26 +1056,26 @@ async def _cog_unpin(self, ctx: commands.Context, *cogs: InstalledCog) -> None:
     async def _cog_listpinned(self, ctx: commands.Context):
         """List currently pinned cogs."""
         installed = await self.installed_cogs()
-        pinned_list = sorted([cog.name for cog in installed if cog.pinned], key=str.lower)
+        pinned_list = sorted([cog for cog in installed if cog.pinned], key=lambda cog: cog.name)
         if pinned_list:
-            message = humanize_list(pinned_list)
+            message = "\n".join(f"(`{cog.commit[:7]}`) {cog.name}" for cog in pinned_list)
```

Comment:
While it is unlikely, there can be cases where `cog.commit` is an empty string (this relates to config schema migration, see GH-2571). In such cases, I think you could use "`unknown`" (or its capitalized form) as the string.